liquidation-monitor/src/agent.spec.js

```diff
@@ -161,125 +231,400 @@ describe('check agent configuration file', () => {
 
 describe('mock axios POST request', () => {
   it('should call axios.post and return a response', async () => {
-    axios.post.mockResolvedValue('test');
     const response = await axios.post('https://...');
     expect(axios.post).toHaveBeenCalledTimes(1);
-    expect(response).toEqual('test');
+    expect(response.data.pagination_summary.total_entries).toEqual(1);
 
     // reset call count for next test
     axios.post.mockClear();
     expect(axios.post).toHaveBeenCalledTimes(0);
   });
 });
 
+// Mock helper functions
+function setDefaultMocks() {
+  // Set all default mocks
+
+  // Compound API
+  axios.post.mockResolvedValue(mockCompoundResponse);
+
+  const {
+    getAssetsIn,
+    markets,
+    getAccountLiquidity,
+    getRateToEth,
+    decimals,
+    getAccountSnapshot,
+    symbol,
+    underlying,
+    exchangeRateStored,
+  } = mockContract;
+
+  // Comptroller
+  /* eslint-disable new-cap */
+  getAssetsIn.mockResolvedValue(mockGetAssetsIn);
+  markets.mockResolvedValue([true, mockBtcCollateralFactor, true]);
+  getAccountLiquidity.mockResolvedValue([mockEthersZero, mockEthersZero, mockEthersZero]);
+  // OneInch
+  getRateToEth.mockResolvedValue(mockBtcPrice);
+  // ERC20
+  decimals.mockResolvedValue(mockBtcDecimals);
+  getAccountSnapshot.mockResolvedValue(
+    [mockEthersZero, mockEthersZero, mockEthersZero, mockEthersZero],
+  );
+  symbol.mockResolvedValue('TOKEN');
+  underlying.mockResolvedValue('0x0');
+  exchangeRateStored.mockResolvedValue(mockBtcCTokenRate);
+
+  // Clear Mock counters before calling initialize
+  axios.post.mockClear();
+  symbol.mockClear();
+  underlying.mockClear();
+  exchangeRateStored.mockClear();
+  decimals.mockClear();
+  getAssetsIn.mockClear();
+}
+
+function setVerifyTokenMocks(setSymbol, setUnderlying, setExchange, setDecimals) {
+  // Verify token MockOnce - BTC then ETH
+  const {
+    decimals,
+    symbol,
+    underlying,
+    exchangeRateStored,
+  } = mockContract;
+
+  symbol.mockResolvedValueOnce(setSymbol);
+  underlying.mockResolvedValueOnce(setUnderlying);
+  exchangeRateStored.mockResolvedValueOnce(setExchange);
+  decimals.mockResolvedValueOnce(mockCDecimals)
+    .mockResolvedValueOnce(setDecimals);
+}
+
 // agent tests
+describe('initializeData', () => {
+  let initializeData;
+
+  beforeEach(async () => {
+    // Initialize
+    initializeData = {};
+    setDefaultMocks();
+    setVerifyTokenMocks('cBTC', '0x0wbtc', mockBtcCTokenRate, mockBtcDecimals);
+    setVerifyTokenMocks('cETH', '0x0weth', mockEthCTokenRate, mockEthDecimals);
+    await (provideInitialize(initializeData))();
+  });
+
+  it('should use axios 2 times}', async () => {
+    // Check counter from the initialize step.
+    expect(axios.post).toBeCalledTimes(2);
+  });
+
+  it('should use contract calls}', async () => {
```

Comment:
```suggestion
  it('should use contract calls', async () => {
```

Reply:
Fixed all instances of this in commit 7717368c93664a46254a77ad97ed46aebd8f3dcf